types/dicer/index.d.ts

```diff
@@ -0,0 +1,173 @@
+// Type definitions for dicer 0.2
+// Project: https://github.com/mscdex/dicer
+// Definitions by: Robin Goupil <https://github.com/GoupilRobin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+/// <reference types="node" />
+import stream = require("stream");
+
+export interface DicerConfig {
+    /**
+     * This is the boundary used to detect the beginning of a new part.
+     */
+    boundary?: string;
+
+    /**
+     * If true, preamble header parsing will be performed first.
+     */
+    headerFirst?: boolean;
+
+    /**
+     * The maximum number of header key=>value pairs to parse Default: 2000 (same as node's http).
+     */
+    maxHeaderPairs?: number;
+}
+
+/**
+ * A very fast streaming multipart parser for node.js.
+ * Dicer is a WritableStream
+ *
+ * Dicer (special) events:
+ * - on('finish', ()) - Emitted when all parts have been parsed and the Dicer instance has been ended.
+ * - on('part', (stream: PartStream)) - Emitted when a new part has been found.
+ * - on('preamble', (stream: PartStream)) - Emitted for preamble if you should happen to need it (can usually be ignored).
+ * - on('trailer', (data: Buffer)) - Emitted when trailing data was found after the terminating boundary (as with the preamble, this can usually be ignored too).
+ */
+export default class Dicer extends stream.Writable {
```

Comment:
The actual export is `module.exports = Dicer`. The equivalent declaration in TypeScript would be `export = Dicer`.

Reply:
AFAIK, using the `export =` notation will force to use `import Dicer = require('dicer')`, while `export default` is much more flexible as it allows the above syntax and other imports syntax, which is nice for people with a preference for the import syntax they use.

I'm fine nesting the interfaces in the `Dicer` class (`DicerConfig` would become `Dicer.Config` and `PartStream` would become `Dicer.PartStream`), but if there is no functionnal difference between the two forms, I would argue that the more flexibility, the better.